src/Operations.jl

```diff
@@ -198,20 +198,38 @@ function collect_fixed!(env, pkgs, uuids)
     fix_deps_map = Dict{UUID,Vector{PackageSpec}}()
     uuid_to_pkg = Dict{UUID,PackageSpec}()
     for pkg in pkgs
-        !has_path(pkg) && continue
-        # A package with a path should have a version
-        @assert pkg.version != nothing
-
-        uuid_to_pkg[pkg.uuid] = pkg
-        # Load the dependencies if this package has a REQUIRE
-        reqfile = joinpath(pkg.path, "REQUIRE")
-        fix_deps_map[pkg.uuid] = valtype(fix_deps_map)()
-        !isfile(reqfile) && continue
-        for r in filter!(r->r isa Pkg2.Reqs.Requirement, Pkg2.Reqs.read(reqfile))
-            pkg_name, vspec = r.package, VersionSpec(VersionRange[r.versions.intervals...])
-            deppkg = PackageSpec(pkg_name, vspec)
-            push!(fix_deps_map[pkg.uuid], deppkg)
-            push!(fix_deps, deppkg)
+        path    = nothing
+        version = nothing
+        has_path(pkg) && (path = pkg.path)
+        has_version(pkg) && (version = pkg.version)
+        info = manifest_info(env, pkg.uuid)
+        version == nothing && info != nothing && haskey(info, "version") && (version = VersionNumber(info["version"]))
+        path == nothing && info != nothing && haskey(info, "path") && (path = info["path"])
+        if path != nothing # This is a fixed package
+            # Package is actually being freed, so it is not fixed
+            pkg.beingfreed && continue
+            # A package with a path should have a version
+            @assert version != nothing
+            pkg.path = path
+            pkg.version = version
+
+            uuid_to_pkg[pkg.uuid] = pkg
+            # Load the dependencies if this package has a REQUIRE
+            # TODO: this is still using Pkg2 facilities
+            reqfile = joinpath(path, "REQUIRE")
+            fix_deps_map[pkg.uuid] = valtype(fix_deps_map)()
+            !isfile(reqfile) && continue
+            for r in filter!(r->r isa Pkg2.Reqs.Requirement, Pkg2.Reqs.read(reqfile))
+                # @assert length(r.versions.intervals) == 1
+                pkg_name, version = r.package, r.versions.intervals[1]
```

Comment:
I had fixed this (and the following lines) in https://github.com/JuliaLang/Pkg3.jl/pull/83/commits/bcdfe81c34bde723681f91f3ede2a62ad46329d1

Reply:
Sorry, must have messed up in rebasing :/

Reply:
It would be preferable to apply the fixes to https://github.com/JuliaLang/Pkg3.jl/pull/83/commits/c728e4bcaa2ae471792482281cb2fd9f14e3d8c1, where the rebase went wrong

Reply:
alright, I'll try...

Reply:
Hopefully good now

Reply:
:+1: